### ISSUE TITLE
atomic species wrong in md.frames

### ISSUE THREAD

**Issue by negf** [OrderN/CONQUEST-release#70]
I assume that the 2 column in md.frames for position (force, velocity) is the atomic species and that the atoms are not reordered during the MD run so the atomic species in md.frames should be identical to what is given initially in coord.in, however they are not identical for all atoms. Moreover, comparing md.frames and trajectory.xsf  the coordinates are identical, while the atomic species differs for some atoms.  Also if I look at the structure of md.frames snapshots they clearly make no sense, while the output in trajectory.xsf looks reasonable. 

The version I used here is Git Branch: master; tag, hash: 1.0.3-pre-1-g386a7c. I also attached some sample output. 
[md.tar.gz](https://github.com/OrderN/CONQUEST-release/files/4941111/md.tar.gz)

**Comment by negf** [comment]
I think it is actually not only the output but the forces get accordingly also scaled by the wrong masses and thus the velocities are not calculated correctly. 

And another odd thing is if atoms are fixed in the coordinate file then the forces are set to 0 zero and they don't move during the MD run, but  the velocities are actually not 0 so they have a finite kinetic energy (temperature), does this make sense?

**Comment by davidbowler** [comment]
There are two problems here which have been located by @tsuyoshi38 relating to atom ordering.  (Conquest uses two approaches to atom ordering: partition labelling (related to parallelisation, which changes during a run as atoms move); and global labelling, which is fixed.)  The first problem is in the subroutine `dump_frame`, in the lines:

```fortran
     write(lun,'(a)') "positions"
     call mdl%dump_mdl_atom_arr(lun, mdl%atom_coords)
     write(lun,'(a)') "end positions"
     write(lun,'(a)') "velocities"
     call mdl%dump_mdl_atom_arr(lun, mdl%atom_velocity)
     write(lun,'(a)') "end velocities"
     write(lun,'(a)') "forces"
     call mdl%dump_mdl_atom_arr(lun, mdl%atom_force)
     write(lun,'(a)') "end forces”
```

found here: https://github.com/OrderN/CONQUEST-release/blob/40cf2904315115b8fd38764b7f493b67793fd935/src/md_model_module.f90#L444-L452

For historical reasons (which will be fixed) ``mdl%atom_velocity`` uses partition labelling while the other arrays use global labelling.  This results in the wrong ordering of the velocities.

The second problem is the association of the wrong species array with the ``mdl`` type (it uses ``species`` where it should used ``species_glob``) here:

https://github.com/OrderN/CONQUEST-release/blob/40cf2904315115b8fd38764b7f493b67793fd935/src/md_model_module.f90#L165

**Comment by tsuyoshi38** [comment]
I have fixed the problem in `dump_frame`.  I also found a bug in `init_velocity`, which gave some initial velocity even for fixed atoms. This can be fixed easily. But, related to this, I would like to discuss one point.

At present, center of mass is fixed as a default setting. But, if we want to fix some of the atoms, I think we should not fix the center of mass. Is it correct ?  And..  even when only one component of an atom is fixed (though I cannot imagine when we need such conditions), is it okay to assume that `flag_FixCOM` is false?
